src/Illuminate/Foundation/Application.php

```diff
@@ -1128,7 +1128,7 @@ public function isLocale($locale)
     public function registerCoreContainerAliases()
     {
         foreach ([
-            'app'                  => [\Illuminate\Foundation\Application::class, \Illuminate\Contracts\Container\Container::class, \Illuminate\Contracts\Foundation\Application::class,  \Psr\Container\ContainerInterface::class],
+            'app'                  => [self::class, \Illuminate\Contracts\Container\Container::class, \Illuminate\Contracts\Foundation\Application::class,  \Psr\Container\ContainerInterface::class],
```

Comment:
This seem to be a breaking change. If for example I'm extending `Illuminate\Foundation\Application` as `App\CoreApplication` it now only can resolve DI for `App\CoreApplication` and not `Illuminate\Foundation\Application`.

Reply:
@crynobone probably needs to be `static::`? Not sure why StyleCI is suddenly changing this. 

Ping @GrahamCampbell

Reply:
As for me all okay ;) currently self always return `\Illuminate\Foundation\Application::class`

and `\Illuminate\Foundation\Application::class` also will return always  `\Illuminate\Foundation\Application::class`